src/components/protection/AddProtectionSingle.vue

```diff
@@ -416,39 +415,27 @@ export default class AddProtectionSingle extends BaseComponent {
     });
   }
 
-  async loadAvailableSpace() {
-    this.loadingAvailableSpace = true;
-    this.amountToMakeSpace = await vxm.ethBancor.getAvailableSpace({
-      poolId: this.pool.id
-    });
-    this.loadingAvailableSpace = false;
-  }
-
-  async loadMaxStakes() {
-    if (this.loadingMaxStakes) return;
-    this.loadingMaxStakes = true;
-    try {
-      const result = await vxm.ethBancor.getMaxStakesView({
-        poolId: this.pool.id
-      });
-      result.forEach(async x => {
-        if (x.token === this.token.symbol) {
-          this.maxStakeAmount = x.amount;
-          this.maxStakeSymbol = x.token;
-        }
-      });
-    } catch (e) {
-      console.log(e);
-    } finally {
-      this.loadingMaxStakes = false;
-    }
-  }
+  async load() {
+    if (this.loading) return;
+    this.loading = true;
 
-  async fetchTknSpaceAvailable() {
-    const result = await vxm.ethBancor.getMaxStakesView({
+    const res = await vxm.ethBancor.getAvailableAndAmountToGetSpace({
       poolId: this.pool.id
     });
-    this.tknSpaceAvailableLTOne = new BigNumber(result[1].amount).lt(1);
+    this.tknSpaceAvailableLTOne = res[0] as boolean;
+    const availableSpace = res[1] as {
+      amount: string;
+      token: string;
+    }[];
+    availableSpace.forEach(x => {
+      if (x.token === this.token.symbol) {
+        this.maxStakeAmount = x.amount;
+        this.maxStakeSymbol = x.token;
+      }
+    });
```

Comment:
From my understanding here, we're looking for the focused token in the available space, once we find it we're updating the view layer vars. 

In the event this array is 2,000 elements long, it's going to keep going even if the ```if``` condition was satisfied on the first element. 
Because we're expecting the focused token to be there, lets do a ```findOrThrow``` which will both ensure the array search will terminate once we've found what we're looking for and if it fails to find, then throw a nice obvious error message for us to tackle. 

```suggestion
    const focusedToken = findOrThrow(availableSpace, space => compareString(space.token, this.tokenSymbol), 
    'failed finding focused token in available space')
     this.maxStakeAmount = focusedToken.amount;
     this.maxStakeSymbol = focusedToken.token;
```

```focusedToken``` might not be the best name here, what do you think better suits?